File: 12-02-2024_Majority_Element.py
from collections import Counter

class Solution:
    def majorityElement(self, nums):
        ele = nums[0]
        count = 1
        n = len(nums)
        for i in range(1, n):
            if count == 0:
                ele = nums[i]
                count += 1
            elif ele == nums[i]:
                count += 1
            else:
                count -= 1
        return ele

File: 12-02-2024_Majority_Element.java
import java.util.List;

public class Solution {
    public int majorityElement(List<Integer> nums) {
        
        int ele = nums.get(0);
        int count = 1;
        int n = nums.size();
        for (int i = 1; i < n; i++) {
            if (count == 0) {
                ele = nums.get(i);
                count++;
            } else if (ele == nums.get(i)) {
                count++;
            } else {
                count--;
            }
        }
        return ele;
    }
}


